src/D2L.Bmx/Okta/Models/AuthenticateResponse.cs

```diff
@@ -2,27 +2,25 @@
 
 namespace D2L.Bmx.Okta.Models;
 
-internal record AuthenticateResponseInital(
+internal abstract record AuthenticateResponse {
+	public record MfaRequired( string StateToken, OktaMfaFactor[] Factors ) : AuthenticateResponse;
+	public record Success( string SessionToken ) : AuthenticateResponse;
+}
+
+internal record AuthenticateResponseRaw(
 	DateTimeOffset ExpiresAt,
 	[property: JsonConverter( typeof( JsonStringEnumConverter ) )]
 	AuthenticationStatus Status,
 	string? StateToken,
 	string? SessionToken,
 	[property: JsonPropertyName( "_embedded" )]
-	AuthenticateResponseEmbeddedInitial? Embedded
+	AuthenticateResponseEmbedded? Embedded
 );
 
-internal record AuthenticateResponseSuccess(
-	DateTimeOffset ExpiresAt,
-	[property: JsonConverter( typeof( JsonStringEnumConverter ) )]
-	AuthenticationStatus Status,
-	string? SessionToken
```

Comment:
Both the primary `authn` call and the MFA `authn/factors/${factorId}/verify` call return the same [authentication transaction object](https://developer.okta.com/docs/reference/api/authn/#authentication-transaction-object). In particular, both of them can return a success status. So it's convenient to define their return types in a polymorphic fashion.

Because STJ's [polymorphic deserialization](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/polymorphism) can't handle this scenario automatically, we will directly deserialize responses to `AuthenticateResponseRaw` first (with all properties included but most nullable), then manually create "tighter" polymorphic objects of type `AuthenticateResponse` depending on values of the raw object.

Related changes:
https://github.com/Brightspace/bmx/pull/313/files#diff-8b1e253817f43a932b584d4f407e885d9f64a848f09317feed09e86d5c3aa1cdR60-R78
https://github.com/Brightspace/bmx/pull/313/files#diff-8b1e253817f43a932b584d4f407e885d9f64a848f09317feed09e86d5c3aa1cdR107-R111